cub/cub/device/dispatch/tuning/tuning_scan.cuh

```diff
@@ -327,13 +754,40 @@ struct policy_hub
     using ScanPolicyT = decltype(select_agent_policy<sm90_tuning<AccumT, is_primitive_op<ScanOpT>()>>(0));
   };
 
-  using MaxPolicy = Policy900;
+  struct Policy1000 : ChainedPolicy<1000, Policy1000, Policy900>
+  {
+    // Use values from tuning if a specialization exists, otherwise pick Policy900
+    template <typename Tuning>
+    static auto select_agent_policy100(int)
+      -> AgentScanPolicy<Tuning::threads,
+                         Tuning::items,
+                         AccumT,
+                         Tuning::load_algorithm,
+                         Tuning::load_modifier,
+                         Tuning::store_algorithm,
+                         BLOCK_SCAN_WARP_SCANS,
+                         MemBoundScaling<Tuning::threads, Tuning::items, AccumT>,
+                         typename Tuning::delay_constructor>;
+    template <typename Tuning>
+    static auto select_agent_policy100(long) -> typename Policy900::ScanPolicyT;
+
+    // Only consider sm100 tunings if the accumulator size matches the one we use in the benchmarks
+    using benchmark_accum_t                = ::cuda::std::__accumulator_t<ScanOpT, ValueT, ValueT>;
+    static constexpr bool accum_size_match = classify_accum_size<AccumT>() == classify_accum_size<benchmark_accum_t>();
+
+    using ScanPolicyT = ::cuda::std::conditional_t<
+      accum_size_match,
+      decltype(select_agent_policy100<sm100_tuning<ValueT, AccumT, OffsetT, classify_op<ScanOpT>()>>(0)),
+      typename Policy900::ScanPolicyT>;
```

Comment:
Here, we check whether the `AccumT` matches what we would use in the benchmark. If it does, we take a `sm100_tuning`, otherwise we fallback to whatever `Policy900` did.